lib/xsv/sheet.rb

```diff
@@ -83,6 +83,13 @@ def [](number)
     # @return [self]
     def parse_headers!
       @headers = parse_headers
+
+      # Check for duplicate headers, but don't care about nil columns
+      if (unique_headers = @headers.compact.uniq) != @headers.compact
+        first_dupe = @headers.compact.zip(unique_headers).detect { |k,v| k != v }.first
+        raise Xsv::DuplicateHeaders, "Duplicate header '#{first_dupe}' found, consider parsing this sheet in array mode."
+      end
```

Comment:
We could simplify this quite a bit, I think.

```
first_duplicate = @headers.detect { |header| @headers.count(header) > 1 }

if (first_duplicate)
  raise Xsv::DuplicateHeaders, "Duplicate header '#{first_duplicate}' found, consider parsing this sheet in array mode."
end
```

Reply:
Thanks, much better